src/vunnel/providers/mariner/parser.py

```diff
@@ -0,0 +1,196 @@
+        self.objects_by_id = {}
+        for obj_element in etree.XPath("//linux-def:rpminfo_object", namespaces=nsmap)(root):
+            try:
+                element_bytes = etree.tostring(obj_element)
+                obj = xml_parser.from_string(element_bytes.decode(), RpminfoObject)
+                self.objects_by_id[obj.id] = obj
+            except Exception as ex:
+                self.logger.warning(f"skipping rpminfo_object element in {oval_file_path} due to {ex}")
+
+        self.states_by_id = {}
+        for state_element in etree.XPath("//linux-def:rpminfo_state", namespaces=nsmap)(root):
+            try:
+                element_bytes = etree.tostring(state_element)
+                state_obj = xml_parser.from_string(element_bytes.decode(), RpminfoState)
+                self.states_by_id[state_obj.id] = state_obj
+            except Exception as ex:
+                self.logger.warning(f"skipping rpminfo_object element in {oval_file_path} due to {ex}")
+        self.mariner_version = oval_file_path.split("-")[-2]  # TODO: less brittle way?
+
+    def name_and_version(self, test_id: str) -> tuple[str | None, str | None]:
+        test = self.tests_by_id.get(test_id, None)
+        if test is None or test.object_value is None or test.state is None:
+            return (None, None)
+        obj = self.objects_by_id.get(test.object_value.object_ref, None)
+        if obj is None:
+            return (None, None)
+        state = self.states_by_id.get(test.state.state_ref, None)
+        if state is None or state.evr is None:
+            return (None, None)
+        return (obj.name, state.evr.value)
+
+    def namespace_name(self) -> str:
+        return f"mariner:{self.mariner_version}"
+
+    def get_test(self, definition: Definition) -> RpminfoTest | None:
+        if definition is None or definition.criteria is None or definition.criteria.criterion is None:
+            return None
+        return self.tests_by_id.get(definition.criteria.criterion.test_ref, None)
+
+    def get_state(self, definition: Definition) -> RpminfoState | None:
+        test = self.get_test(definition)
+        if test is None or test.state is None or test.state.state_ref is None:
+            return None
+        return self.states_by_id.get(test.state.state_ref, None)
+
+    def get_object(self, definition: Definition) -> RpminfoObject | None:
+        test = self.get_test(definition)
+        if test is None or test.object_value is None or test.object_value.object_ref is None:
+            return None
+        return self.objects_by_id.get(test.object_value.object_ref, None)
+
+    def make_fixed_in(self, definition: Definition) -> FixedIn | None:
+        state = self.get_state(definition)
+        obj = self.get_object(definition)
+        if state is None or state.evr is None:
+            return None
+        if obj is None or obj.name is None:
+            return None
+        version = state.evr.value
+        if state.evr.operation == LTE:
```

Comment:
this is fairly tricky, a healthy comment here explaining why `LTE` means we should consider all versions vulnerable would be helpful for future devs here.

Reply:
How's this?

> There are 2 choices for state.ever.operation: "less than" or "less than or equal to".
So for each vulnerability, the data either says "versions < 3.2.1 are vulernable"
or the statement "versions <= 3.2.1 are vulnerable". In the second statement,
the data has no information about any fixed version, so we report "None"
as the fixed version, meaning we consider all version vulnerable.
For example, if version 3.2.1 of a library is vulnerable, and is the latest version
mariner data might have "versions <= 3.2.1" is vulnerable. If the library is then fixed in version 3.3.1, the data will be updated to say "versions < 3.3.1", and we can infer that it was fixed in 3.3.1.